### PR TITLE
Live sessions list

### PR DESCRIPTION
**Pull request by bmarty**
When another session was remotely deleted (i.e. signed out), the session list was not updating correctly.

Fix other RX similar issues.

First commit is just a renaming

### CODE REVIEW

File: matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/model/rest/UploadSignatureQueryBuilder.kt [hunk 1]
```diff
@@ -38,11 +38,8 @@ internal data class UploadSignatureQueryBuilder(
     fun build(): Map<String, Map<String, @JvmSuppressWildcards Any>> {
         val map = HashMap<String, HashMap<String, Any>>()
 
-        val usersList = (
-                deviceInfoList.map { it.userId }
-                        + signingKeyInfoList
-                        .map { it.userId }
-                ).distinct()
+        val usersList = (deviceInfoList.map { it.userId } + signingKeyInfoList.map { it.userId })
+                .distinct()
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/crypto/model/rest/UploadSignatureQueryBuilder.kt:42]: This is just formatting

File: vector/src/main/java/im/vector/app/features/settings/devices/DevicesViewModel.kt [hunk 2]
```diff
@@ -137,12 +137,14 @@ class DevicesViewModel @AssistedInject constructor(
 //                }
 
         session.rx().liveUserCryptoDevices(session.myUserId)
-                .distinct()
-                .throttleLast(5_000, TimeUnit.MILLISECONDS)
+                .map { it.size }
+                .distinctUntilChanged()
+                .throttleFirst(5_000, TimeUnit.MILLISECONDS)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/settings/devices/DevicesViewModel.kt:142]: throttle first is better

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/model/rest/UploadSignatureQueryBuilder.kt [hunk 3]
```diff
@@ -39,11 +39,8 @@ internal data class UploadSignatureQueryBuilder(
     fun build(): Map<String, Map<String, @JvmSuppressWildcards Any>> {
         val map = HashMap<String, HashMap<String, Any>>()
 
-        val usersList = (
-                deviceInfoList.map { it.userId }
-                        + signingKeyInfoList
-                        .map { it.userId }
-                ).distinct()
+        val usersList = (deviceInfoList.map { it.userId } + signingKeyInfoList.map { it.userId })
+                .distinct()
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/model/rest/UploadSignatureQueryBuilder.kt:43]: this is just formatting

Comment (BillCarsonFr) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/model/rest/UploadSignatureQueryBuilder.kt:43]: no distinctUntilChange here?

File: vector/src/main/java/im/vector/app/features/settings/devices/DevicesViewModel.kt (hunk 2 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/app/features/settings/devices/DevicesViewModel.kt:142]: I think that I made it to wait a bit mainly for the first time. If not just after a new session you could have verification alerts coming to fast

Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/settings/devices/DevicesViewModel.kt:142]: Ok, I revert

**Pull request merged**